keras/applications/resnet50.py

```diff
@@ -228,10 +227,8 @@ def ResNet50(include_top=True, weights='imagenet',
     x = identity_block(x, 3, [512, 512, 2048], stage=5, block='b')
     x = identity_block(x, 3, [512, 512, 2048], stage=5, block='c')
 
-    x = AveragePooling2D((7, 7), name='avg_pool')(x)
-
     if include_top:
-        x = Flatten()(x)
+        x = GlobalAveragePooling2D()(x)
```

Comment:
It appears `GlobalAveragePooling2D` already flattens:
```
import keras
inputs = keras.layers.Input((224, 224, 3))
x = keras.layers.GlobalAveragePooling2D()(inputs)
print(x.shape)
```
```
Using TensorFlow backend.
(?, 3)
```

It calls `K.mean(inputs, axis=[1, 2])`, meaning the parameter `keepdims` of `tf.nn.reduce_mean` is falls, meaning the dims are dropped altogether.

According to the `GlobalAveragePooling2D` docs:
```
    # Output shape
        2D tensor with shape:
        `(batch_size, channels)`
```